src/fs.rs

```diff
@@ -270,11 +270,11 @@ where
         debug!("no syscall arguments received from /proc/<pid>/syscall");
         return None;
     }
-    // FIXME: We need to allow open/openat because some programs want to open themself, i.e. bash
-    let allowed_syscall = is_open_syscall(args[0])
-        && args[0] == libc::SYS_execve as usize
-        && !env.contains_key(OsStr::new("ENVFS_RESOLVE_ALWAYS"));
+    // We need to allow open/openat because some programs want to open themself, i.e. bash
+    let allowed_syscall =
+        is_open_syscall(args[0]) || env.contains_key(OsStr::new("ENVFS_RESOLVE_ALWAYS"));
```

Comment:
I am currently wondering, but shouldn't this also allow the execve systemcall?

Reply:
There is a special case for handling `execve` few lines below. This is only for handling the rest. I would prefer moving this part few lines below, close to the place where `allowed_syscall` is used. But I didn't want to make "too big" changes to the code.

Reply:
Check out https://github.com/Mic92/envfs/pull/146

Reply:
In the error case execve would still fall through.